server/issue_test.go

```diff
@@ -101,3 +105,47 @@ func TestTransitionJiraIssue(t *testing.T) {
 		})
 	}
 }
+
+func TestRouteIssueTransition(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError",
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string")).Return(nil)
+
+	api.On("LogDebug",
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string")).Return(nil)
+
+	p := Plugin{}
+	p.SetAPI(api)
+
+	p.userStore = getMockUserStoreKV()
+	p.currentInstanceStore = mockCurrentInstanceStore{&p}
+
+	request := httptest.NewRequest("POST", routeIssueTransition, nil)
+	request.Header.Set("Mattermost-User-Id", "connected_user")
+	w := httptest.NewRecorder()
+	p.ServeHTTP(&plugin.Context{}, w, request)
+	assert.Equal(t, 400, w.Result().StatusCode)
```

Comment:
If this test isn't really doing anything except asserting a `400` status. There's not much we can do since we currently don't have Jira API mocking done in our tests. 1/5 I suggest removing the test